docs/proposed/grid-manager/managed-grid.rst

```diff
@@ -0,0 +1,338 @@
+- `--name`: the name of this certificate (default: "default")
+
+Import a "version 1" storage-certificate produced by a grid-manager
+(probably: a storage server may have zero or more such certificates
+installed; for now just one is sufficient). You will have to re-start
+your node after this. Subsequent announcements to the Introducer will
+include this certificate.
+
+.. note::
+
+   This command will simply edit the `tahoe.cfg` file and direct you
+   to re-start. In the Future(tm), we should consider (in exarkun's
+   words):
+
+       "A python program you run as a new process" might not be the
+       best abstraction to layer on top of the configuration
+       persistence system, though.  It's a nice abstraction for users
+       (although most users would probably rather have a GUI) but it's
+       not a great abstraction for automation.  So at some point it
+       may be better if there is CLI -> public API -> configuration
+       persistence system.  And maybe "public API" is even a network
+       API for the storage server so it's equally easy to access from
+       an agent implemented in essentially any language and maybe if
+       the API is exposed by the storage node itself then this also
+       gives you live-configuration-updates, avoiding the need for
+       node restarts (not that this is the only way to accomplish
+       this, but I think it's a good way because it avoids the need
+       for messes like inotify and it supports the notion that the
+       storage node process is in charge of its own configuration
+       persistence system, not just one consumer among many ... which
+       has some nice things going for it ... though how this interacts
+       exactly with further node management automation might bear
+       closer scrutiny).
+
+
+Enrolling a Storage Server: Config
+----------------------------------
+
+You may edit the ``[storage]`` section of the ``tahoe.cfg`` file to
+turn on grid-management with ``grid_management = true``. You then must
+also provide a ``[grid_management_keys]`` section in the config-file which
+lists ``name = path/to/certificate`` pairs.
+
+These certificate files are issued by the ``tahoe grid-manager sign``
+command; these should be **securely transmitted** to the storage
+server. Relative paths are based from the node directory. Example::
+
+    [storage]
+    grid_management = true
+
+    [grid_management_keys]
+    default = example_grid.cert
+
+This will cause us to give this certificate to any Introducers we
+connect to (and subsequently, the Introducer will give the certificate
+out to clients).
+
+
+Enrolling a Client: CLI
+-----------------------
+
+DECIDE: is a command like this best, or should you have to edit the
+        config "by hand"? (below fits into warner's philosophy that "at some
+        point" it might be best to have all config in a database or similar
+        and the only way to view/edit it is via tahoe commands...)
```

Comment:
Okay, I definitely agree with that for most Tahoe things. 

For a bunch of the `tahoe grid-manager` commands I would expect them to be used before any actual nodes exist and therefore there is no network API to call. For this one in particular, a network API would make sense -- however, it would make **even more** sense if that were part of a general "edit the configuration" series of API calls.

I am taking your comment to mean "it's okay for Grid Manager to lean on a CLI for this, but we shouldn't keep doing it"...?

We would also have to firmly agree on the mechanism of authentication for "dangerous" API calls such as this -- that edit the configuration. So far, the WebAPI reveals nothing unless you already have secret information (such as a read-cap). The single exception is the "status" API that was intended to start the conversation about "dangerous" API calls and how to make them "no worse than editing the files in the node-dir".

So, my action here will be to delete the "DECIDE" comment -- when we grow such an API, it will be easy to gut this CLI command and replace it with a Web request.